mod/metric_answer_lookup/lib/answer/answer_class_methods.rb

```diff
@@ -65,11 +65,8 @@ def split_search_args args
 
     # @param ids [Integer, Array<Integer>] card ids of metric answer cards
     def refresh ids=nil, *fields
-      ids = Array(ids).compact
-      if ids.present?
-        ids.each { |ma_id| refresh_entry fields, ma_id }
-      else
-        refresh_all fields
+      Array(ids).compact.each do |ma_id|
+        refresh_entry fields, ma_id
       end
```

Comment:
this, it turns out, was the problem.  if refresh gets no id, it refreshes _everything_.  That's a LOT of answers being refreshed a LOT of times.

Reply:
🙀